test/AuthorStake.t.sol

```diff
@@ -0,0 +1,367 @@
+
+        vm.deal(staker, 1 ether); // Ensure staker has some ether to cover gas costs
+        vm.prank(staker);
+
+        vm.expectRevert(bytes(""));
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+    }
+
+    function testStakingAndWithdrawingInLoops() public {
+        address staker = address(11);
+        uint256 iterations = 5;
+        uint256 stakePerIteration = 0.1 ether;
+        uint256 totalStake = stakePerIteration * iterations;
+
+        vm.deal(staker, totalStake);
+        for (uint256 i = 0; i < iterations; i++) {
+            vm.prank(staker);
+            stakingContract.stakeEtherFrom{value: stakePerIteration}(staker);
+        }
+
+        for (uint256 i = 0; i < iterations; i++) {
+            vm.prank(staker);
+            stakingContract.withdrawStake(stakePerIteration);
+        }
+
+        // Verify the balance is zero after all withdrawals
+        uint256 remainingBalance = stakingContract.getStakedBalance(staker);
+        assertEq(
+            remainingBalance,
+            0,
+            "Staker's balance should be zero after iterative withdrawals"
+        );
+    }
+
+    function testEventEmissionOnStakingAndWithdrawal() public {
+        address staker = address(10);
+        uint256 stakeAmount = 0.4 ether;
+        uint256 withdrawAmount = 0.2 ether;
+
+        vm.recordLogs();
+        vm.deal(staker, stakeAmount);
+        vm.expectEmit(address(stakingContract));
+        vm.startPrank(staker);
+
+        emit EtherStaked(staker, stakeAmount);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+
+        vm.expectEmit(address(stakingContract));
+        emit EtherWithdrawn(staker, withdrawAmount);
+        stakingContract.withdrawStake(withdrawAmount);
+
+        vm.stopPrank();
+    }
+
+    function testFuzzStakeEther(uint256 _amount) public {
+        // Skip the test case if the fuzzed amount is 0 to avoid failing the "Must send Ether to stake" requirement
+        if (_amount == 0) return;
+
+        // Ensure the staker has enough Ether to cover the stake.
+        uint256 stakeAmount = _amount % 10 ether; // Example to limit the stake amount for practical testing
+
+        address staker = address(this);
+        vm.deal(staker, stakeAmount + 1 ether); // Ensure the staker has more than the stake amount
+        vm.prank(staker);
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+
+        uint256 stakedBalance = stakingContract.getStakedBalance(staker);
+        assertEq(
+            stakedBalance,
+            stakeAmount,
+            "Staked amount does not match expected balance"
+        );
+    }
+
+    function testFuzzWithdrawalAmounts(uint256 _withdrawAmount) public {
+        // Ensure there's a non-zero amount to work with, adjusted to avoid excessively large transactions
+        uint256 withdrawAmount = (_withdrawAmount % 1 ether) + 1; // Ensure at least 1 wei is withdrawn and cap at 1 ether for practicality
```

Comment:
This isn't necessary, and we can limit the amount using the **vm.assume** keyword.